test/python/quantum_info/operators/symplectic/test_pauli_table.py

```diff
@@ -0,0 +1,1240 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Tests for PauliTable class."""
+
+import unittest
+from test import combine
+from ddt import ddt
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from qiskit import QiskitError
+from qiskit.test import QiskitTestCase
+from qiskit.quantum_info.operators.symplectic import PauliTable
+
+
+def pauli_mat(label):
+    """Return Pauli matrix from a Pauli label"""
+    mat = np.eye(1, dtype=complex)
+    for i in label:
+        if i == 'I':
+            mat = np.kron(mat, np.eye(2, dtype=complex))
+        elif i == 'X':
+            mat = np.kron(mat, np.array([[0, 1], [1, 0]], dtype=complex))
+        elif i == 'Y':
+            mat = np.kron(mat, np.array([[0, -1j], [1j, 0]], dtype=complex))
+        elif i == 'Z':
+            mat = np.kron(mat, np.array([[1, 0], [0, -1]], dtype=complex))
+        else:
+            raise QiskitError('Invalid Pauli string {}'.format(i))
+    return mat
+
+
+class TestPauliTableInit(QiskitTestCase):
+    """Tests for PauliTable initialization."""
+
+    def test_array_init(self):
+        """Test array initialization."""
+        # Matrix array initialization
+        with self.subTest(msg='bool array'):
+            target = np.array([[False, False], [True, True]])
+            value = PauliTable(target)._array
+            self.assertTrue(np.all(value == target))
+
+        with self.subTest(msg='bool array no copy'):
```

Comment:
It's checking that the input array isn't copied so that any modifications of the array through the class will also modify the underlying numpy array it was initialized with. This is both for efficiency, and is also needed for all the fancy `[]` getter/setters to work. If you do `table[[0, 2, 4]]` for example it will return a 3 row PauliTable object containing the 0, 2, 4th rows of the original, and modifiying them will also change those rows in the original table.